### ISSUE TITLE
Dynamic spawning of walker

### ISSUE THREAD

**Issue by droidXrobot** [Unity-Technologies/ml-agents#5630]
**Describe the bug**
Inference works well in the Walker scene when the agents start with the script, but if you spawn an agent later on they seem out of sync and don't walk well.

**To Reproduce**

1. Deactivate some of the walkers prefabs
2. Start the scene
3. Activate the walkers

Walkers that started with the scene will walk normally, ones that were activated will fall often.
**Console logs / stack traces**
Please wrap in [triple backticks (```)](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) to make it easier to read.

**Environment**
- Unity Version:  2020.2.16f1
- OS + version: Windows 10
- _ML-Agents version_: Verified Package 1.0.8
- _Torch version_: TensorFlow version
- _Environment_: Walker example scene

**Comment by jrupert-unity** [comment]
This bug is confirmed to happen with latest versions for Unity (2020.3.21f1) and ML-Agents (2.0.1) and is being investigated (Jira MLA-2321).

**Comment by mrkgmr** [comment]
This happend to me too but with other versions. 
In our case, we instantiate the prefabs and the we train them. In this point, the trained brain work perfectly in this instantiated prefabs, but if we delete them and we instantiate again they stop to work with this brain. I suspect about some kind of error in the NNModel initialitation or something else ... are there some workaround?

**Environment**

-  Unity Version: 2020.3.18f1
- OS + version: Windows 10
- ML-Agents version: 2.1.0-exp.1
- Torch Version: TensorFlow version
- Environment: Walker example customized

**Comment by mrkgmr** [comment]
Hi again, I did some test, maybe it helps. I start from a scenary with one single environment NO prefab, that works perfectly with my trained model. It's a custom Walker. 

- I turned it in a prefab without removing from hierarchy -> It doesn't work
- I turned it in a prefab without removing from hierarchy + Save project -> **It works**
- Disable in hierarchy + Enable in hierarchy -> It doesn't work
- Disable in hierarchy + Enable in hierarchy + Save project -> **It works**

- I turned it in a prefab without removing from hierarchy + Save project + Remove it from hierarchy and spawn it again using "PrefabUtility.InstantiatePrefab"-> It doesn't work
- I turned it in a prefab without removing from hierarchy + Save project + Remove it from hierarchy and spawn it again using "PrefabUtility.InstantiatePrefab" + Save project -> It doesn't work
- I turned it in a prefab without removing from hierarchy + Save project + Remove it from hierarchy and spawn it again using "Instantiate" -> It doesn't work
- I turned it in a prefab without removing from hierarchy + Save project + Remove it from hierarchy and spawn it again using "Instantiate" + Save project -> It doesn't work

It seems two different problems, but maybe they have a relationship with each other.
I hope this help you to fix it.
